pkg/logs/client/http/sync_destination.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//nolint:revive // TODO(AML) Fix revive linter
+package http
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/logs/client"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/logs/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// SyncDestination sends a payload over HTTP and does not retry.
+type SyncDestination struct {
+	destination *Destination
+}
+
+// NewSyncDestination returns a new synchronous Destination.
+func NewSyncDestination(endpoint config.Endpoint,
```

Comment:
@remeh wanted to see if this was this along the lines of your suggestion? I tried to cut out the concurrent pieces of the typical HTTP Destination